### ISSUE TITLE
Recommended list: IANA or PSL?

### ISSUE THREAD

**Issue by WilliamDEdwards** [jeremykendall/php-domain-parser#353]
Should the IANA or the PSL list be used in production?

The PSL maintainer recommends using the IANA list (see below), but that one does not contain TLDs such as `.co.uk`...

> Using the Public Suffix List to determine what is a valid domain name and what isn't is dangerous, particularly in these days when new gTLDs are arriving at a rapid pace.
> If you are looking to know the validity of a Top Level Domain, the IANA Top Level Domain List is the proper source for this information or alternatively consider using directly the DNS.

(https://github.com/jeremykendall/php-domain-parser)

> PSL maintainer here: please don’t use the PSL!
> [...] if you are going to use a list, using the IANA list, updated daily, is much better than the PSL.

(https://news.ycombinator.com/item?id=24437644)

**Comment by nyamsprod** [comment]
and they are correct PSL and IANA are complementary. the former is about domain suffix while the latter is about registered TLDs. Let's take a quick example 

If on Monday someone register on the IANA list the TLD `foobar` then the IANA list and subsequently the DNS system will be made aware of it as late as tuesday not the PSL one.

But if subsequently the `co.foobar` can be bought on a registrar then it is more than likely that it has been added to the PSL list (not guaranteed).

**TL;DR** the list up to date for TLDs is the IANA one.

The PSL is the maintained list of PSL but there's no guarantee that the list is either accurate or update to date has its update is not done on a regular basis and is "voluntary".

**Comment by WilliamDEdwards** [comment]
To correct myself: `.co.uk` is not a TLD. `.uk` is. `.co` is an SLD.

To your knowledge, is there a properly maintained suffix list? I understand from your message that the IANA list contains only TLDs, and the PSL contains suffixes. However, apparently, the PSL should not be used...

**Comment by nyamsprod** [comment]
The PSL **SHOULD BE USED** but not for validating that a TLD is indeed valid.

**Comment by WilliamDEdwards** [comment]
So using the PSL is not 'dangerous', but could miss TLDs as it is not updated regularly. Perhaps that should be clarified in the README.

**Comment by nyamsprod** [comment]
Well that's it clearly state on the PSL website not sure why it should be said here again 🤔 see (https://publicsuffix.org/learn/).
